roles/foreman_repositories/tasks/redhat.yml

```diff
@@ -36,6 +36,8 @@
 - name: Enable powertools for Katello
   when:
     - foreman_repositories_katello_version is defined
+    - foreman_repositories_katello_version != 'nightly'
+    - foreman_repositories_katello_version is version('4.10', '<')
```

Comment:
Do you have to specify not nightly here?

Reply:
`version()` uses Python's version comparison code, and that does not cope well with versions that are not "numeric", so we filter out `nightly` earlier, so `version()` is never called with it.